token/core/zkatdlog/crypto/common/array.go

```diff
@@ -33,6 +35,7 @@ func (a *G1Array) Bytes() ([]byte, error) {
 			return nil, errors.Errorf("failed to marshal array of G1")
 		}
 		bytes := e.Bytes()
+		raw = append(raw, []byte(Seperator)...)
```

Comment:
General comment - when you consecutively append to an empty slice like that, it's `O(n*log(n))` complexity, because each time the slice's capacity is not enough to hold the required bytes, so Go's runtime allocates a new array twice the size of the old array and copies over the bytes. So it's better to pre-allocate the `raw`'s capacity to be an upper bound on the cumulative size of the elements, and then iterate and append.

Reply:
I will address this in the PR dedicated to cleaning up the code.